packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -124,6 +128,8 @@ identifier.create = (branch: TreeBranch): string => {
 
 Object.freeze(identifier);
 
+export type ParentObject = TreeView<ImplicitFieldSchema> | TreeNode;
```

Comment:
This should have a doc comment noting the semantics of this type. For example that for nodes they are the parents of all their immediate children (all their fields's contents) and for branches they are the parent of the document root node.

It should clarify what the keys are for the special cases (branches) and how removed and unhydrated nodes are handled. Assuming we plan to add support for them later, we likley will want a disclaimer on this type (as private remakrs) noting more values can be added to it and this should be done before stabilizing it.